src/testing.rst

```diff
@@ -29,8 +34,125 @@ On Windows, use the ``exist`` command. See below for an example.
         - if not exist %PREFIX%\\Library\\lib\\boost_log-vc140-mt.lib exit 1  # [win and py35]
 
 
+Testing Python Packages
+-----------------------
+
+Testing Importing
+.................
+
+The minimal test of a python package should make sure that the package imports. This can
+be accomplished with this stanza in the yaml:
+
+.. code-block:: yaml
+
+    test:
+      imports:
+        - package_name
+
+Note that that ``package_name`` is the name that is imported by python,
+not necessarily the name of the conda package (they are sometimes different).
+
+Testing for an import will catch the bulk of the packaging errors, generally
+including presence of dependencies. However, it does not assure that the
+package works correctly particularly if it works correctly with the particular
+versions of dependencies used. So it is good to run some other test of the code itself.
+
+Running Unit Tests
+..................
+
+The trick here is that there are multiple ways to run unit tests in Python,
+including nose, pytest, etc.
+
+Also, some packages install the tests with the package, and thus they can be
+run in place. while others keep the tests with the source code, and thus can
+not be run straight from an installed package.
+
+Test Requirements
+.................
+
+Sometimes there are packages required to run the tests that are not required
+to simply use the package. This is usually a test-running framework, such as
+nose or pytest. You can ensure that it is included by adding it to the test
+stanza:
+
+.. code-block:: yaml
+
+    test:
+      imports:
+        - package_name
+    ...
+      requires:
+        - pytest
+
+Built-in Tests
+..............
+
+Some packages have testing built-in. In this case, you can put a test command
+directly in the test stanza:
+
+.. code-block:: yaml
+
+    test:
+      ...
+      commands:
+         python -c "import package_name; package_name.tests.runall()"
+
+Alternatively, you can add a file called ``run_test.py`` in the recipe that
+will be run at test time. This allows an arbitrarily complicated test script.
+
+pytest Tests
+............
+
+If the tests are installed with the package, pytest can find and run them
+for you with the following command::
+
+    test:
+      requires:
+        - pytest
+      commands:
+        - py.test --pyargs package_name
+
+nose Tests
+...........
+
+Someone please add a description and example for running nose on an installed package.
```

Comment:
Basically the same store, but add `nose` to `requires` and change the command to `nosetests package_name`.